src/main/java/uk/nhs/ctp/tkwvalidation/ZipBuilderFactory.java

```diff
@@ -0,0 +1,11 @@
+package uk.nhs.ctp.tkwvalidation;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ZipBuilderFactory {
+
+  public ZipBuilder create() {
+    return new ZipBuilder();
+  }
```

Comment:
You could simplify the singleton instance here by making this a config:
```suggestion
@Configuration
public class ZipBuilderConfig {
  @Bean
  public ZipBuilder create() {
    return new ZipBuilder();
  }
```
Then just autowire in the zip builder and no need to mock 2 things in the test